### PR TITLE
Persist name resolution for class measures

### PR DESCRIPTION
**Pull request by facundominguez**
Another step for #2169

### CODE REVIEW

File: liquidhaskell-boot/src/Language/Haskell/Liquid/LHNameResolution.hs [hunk 1]
```diff
@@ -682,6 +685,16 @@ resolveLogicNames cfg env globalRdrEnv unhandledNames lmap0 localVars lnameEnv p
     findReflection :: GHC.Name -> Maybe LHName
     findReflection n = GHC.lookupNameEnv (lneReflected lnameEnv) n
 
+mapMeasureNamesM :: Monad m => (LocSymbol -> m LocSymbol) -> MeasureV v ty ctor -> m (MeasureV v ty ctor)
```
Comment (sjoerdvisscher) [inline liquidhaskell-boot/src/Language/Haskell/Liquid/LHNameResolution.hs:688]: This is smelling a lot like lenses!

File: tests/errors/MultiInstMeasures.hs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{-@ LIQUID "--expect-error-containing=Multiple instance measures `sizeOf` for type `GHC.Internal.Ptr.Ptr`" @-}
+{-@ LIQUID "--expect-error-containing=Multiple instance measures `MultiInstMeasures.sizeOf` for type `GHC.Internal.Ptr.Ptr`" @-}
```
Comment (sjoerdvisscher) [inline tests/errors/MultiInstMeasures.hs:1]: I would not expect a qualified name for an identifier from the current module.

**Review by sjoerdvisscher — APPROVED** [review]
Less comments to check the code against, but I guess that makes sense since nothing out of the ordinary is happening.

File: tests/errors/MultiInstMeasures.hs (hunk 2 above)
Comment (facundominguez) [reply tests/errors/MultiInstMeasures.hs:1]: It is an artifact of renaming. Names are qualified now, and I'm guessing the tidy passes for errors would need to be updated to strip the qualifiers or replace them with import aliases.

I'm leaving that for another time, but feel free to object.

File: liquidhaskell-boot/src/Language/Haskell/Liquid/LHNameResolution.hs (hunk 1 above)
Comment (facundominguez) [reply liquidhaskell-boot/src/Language/Haskell/Liquid/LHNameResolution.hs:688]: Yeah. I'm not eager to introduce the lense machinery for a few queries. But I'm open to be convinced that there are sufficient use opportunities to be worth it.

**Comment by facundominguez** [comment]
Many thanks for the review @sjoerdvisscher!

**Pull request merged**